### ISSUE TITLE
Core provider methods

### ISSUE THREAD

**Issue by krsriq** [Nyholm/next-faker#9]
Right now there are a couple of methods added to the core providers, as suggested in https://github.com/Nyholm/next-faker/blob/master/PROVIDERS.md, such as `state` and `secondaryAddress` (both currently not in `Faker\Provider\Address`).

I've made a [little analysis](https://gist.github.com/krsriq/efff59f0e47bfe11d20c0cfdfa3a8263) to see what methods are actually present in localized providers and whether they are currently in the core providers - here's a snippet:

| method name | in localized providers count | in core providers |
| --- | --- | --- | 
| bankAccountNumber | 43 | ✖ |
| streetPrefix | 30 | ✖ |
| cityName | 29 | ✖ |
| cityPrefix | 23 | ✖ |
| state | 23 | ✖ |
| mobileNumber | 20 | ✖ |
| secondaryAddress | 19 | ✖ |
| city | 18 | ✔ |

Based on this 

- how about we add `bankAccountNumber` to the core provider `Payment`? (btw - is payment renamed in the provider suggestion to money on purpose?)
- how about we add `streetPrefix` to the core provider `Address` - this one's not yet in `en_US`
- how about we add `mobileNumber` to the core provider `PhoneNumber` - this one's not yet in `en_US`

**Comment by Nyholm** [comment]
FYI @pimjansen

**Comment by pimjansen** [comment]
Thanks for the input. Is it time to create the interfaces on what can be added in core?

My approach would be to keep everything there "as is" unless it is really weird. All other stuff we can deprecate since else im afraid that the transition is a hell. 

One of the better things of Faker atm is the ease of use so.

What dk you think?

**Comment by Nyholm** [comment]
I agree that we should keep things "as is" and have a smooth migration path. 
If we look at the research from Daniel, it seams like most things implement backAccrountNumber, street prefix etc. 

I also thinks some more of this could be removed/deprecated, ie `toUpper`. 

Since we are creating new interfaces here, we dont have to focus on our legacy. The legacy methods should still work of course, but they should not be added to a new interface. 

Let's use this `PROVIDERS.md` document to do some changes and shuffle things around before we start writing code.

**Comment by pimjansen** [comment]
Agree!

Also lets make some proposals to simplify since streetName can easily be called street from the interface.

One of the issues Faker 1 has is that all methods are called magicly and therefor conflict after beging merged as 1 class itself
